lib/ansible/modules/cloud/azure/azure_rm_storageaccount.py

```diff
@@ -175,34 +175,122 @@
 
 RETURN = '''
 state:
-    description: Current state of the storage account.
+    description:
+        - Current state of the storage account.
     returned: always
-    type: dict
-    sample: {
-        "account_type": "Standard_RAGRS",
-        "custom_domain": null,
-        "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/clh0003",
-        "location": "eastus2",
-        "name": "clh0003",
-        "primary_endpoints": {
-            "blob": "https://clh0003.blob.core.windows.net/",
-            "queue": "https://clh0003.queue.core.windows.net/",
-            "table": "https://clh0003.table.core.windows.net/"
-        },
-        "primary_location": "eastus2",
-        "provisioning_state": "Succeeded",
-        "resource_group": "Testing",
-        "secondary_endpoints": {
-            "blob": "https://clh0003-secondary.blob.core.windows.net/",
-            "queue": "https://clh0003-secondary.queue.core.windows.net/",
-            "table": "https://clh0003-secondary.table.core.windows.net/"
-        },
-        "secondary_location": "centralus",
-        "status_of_primary": "Available",
-        "status_of_secondary": "Available",
-        "tags": null,
-        "type": "Microsoft.Storage/storageAccounts"
-    }
+    type: complex
+    contains:
+        account_type:
+            description:
+                - Type of storage account.
+            returned: always
+            type: str
+            sample: Standard_RAGRS
+        custom_domain:
+            description:
+                - User domain assigned to the storage account.
+            returned: always
+            type: complex
+            contains:
+                name:
+                    description:
+                        - CNAME source.
+                    returned: always
+                    type: str
+                    sample: testaccount
+                use_sub_domain:
+                    description:
+                        - Whether to use sub domain.
+                    returned: always
+                    type: bool
+                    sample: true
+        id:
+            description:
+                - Resource ID.
+            returned: always
+            type: str
+            sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/clh0003"
+        location:
+            description:
+                - Valid Azure location. defaults to location of the resource group.
```

Comment:
Missed this one in my first review:

```suggestion
                - Valid Azure location. Defaults to location of the resource group.
```